cmd/astrolabe/main.go

```diff
@@ -0,0 +1,312 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+	"github.com/vmware-tanzu/astrolabe/gen/client"
+	"github.com/vmware-tanzu/astrolabe/pkg/astrolabe"
+	client2 "github.com/vmware-tanzu/astrolabe/pkg/client"
+	"github.com/vmware-tanzu/astrolabe/pkg/server"
+	"io"
+	"log"
+	"os"
+)
+
+func main() {
+	app := &cli.App{
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "host",
+				Usage: "Astrolabe server",
+			},
+			&cli.BoolFlag{
+				Name:     "insecure",
+				Usage:    "Only use HTTP",
+				Required: false,
+				Hidden:   false,
+				Value:    false,
+			},
+			&cli.StringFlag{
+				Name:  "confDir",
+				Usage: "Configuration directory",
+			},
+		},
+		Commands: []*cli.Command{
+			{
+				Name:   "types",
+				Usage:  "shows Protected Entity Types",
+				Action: types,
+			},
+			{
+				Name:   "ls",
+				Usage:  "lists entities for a type",
+				Action: ls,
+			},
+			{
+				Name:      "show",
+				Usage:     "shows info for a protected entity",
+				ArgsUsage: "<protected entity id>",
+				Action:    show,
+			},
+			{
+				Name:      "lssn",
+				Usage:     "lists snapshots for a Protected Entity",
+				Action:    lssn,
+				ArgsUsage: "<protected entity id>",
+			},
+			{
+				Name:      "snap",
+				Usage:     "snapshots a Protected Entity",
+				Action:    snap,
+				ArgsUsage: "<protected entity id>",
+			},
+			{
+				Name:      "rmsn",
+				Usage:     "removes a Protected Entity snapshot",
+				Action:    rmsn,
+				ArgsUsage: "<protected entity snapshot id>",
+			},
+			{
+				Name:      "cp",
+				Usage:     "copies a Protected Entity snapshot",
+				Action:    cp,
+				ArgsUsage: "<src> <dest>",
+			},
+		},
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func setupProtectedEntityManager(c *cli.Context) (astrolabe.ProtectedEntityManager, error) {
+	confDirStr := c.String("confDir")
+	var pem astrolabe.ProtectedEntityManager
+	if confDirStr != "" {
+		pem = server.NewProtectedEntityManager(confDirStr)
+	}
+	host := c.String("host")
+	if host != "" {
+		insecure := c.Bool("insecure")
+		transport := client.DefaultTransportConfig()
+		transport.Host = host
+		if insecure {
+			transport.Schemes = []string{"http"}
+		}
+
+		restClient := client.NewHTTPClientWithConfig(nil, transport)
+		var err error
+		pem, err = client2.NewClientProtectedEntityManager(restClient)
```

Comment:
`pem, err := ...` would be easier

Reply:
Yes, it would be.  I did this because of a goism.  err was not declared outside the block but pem was.  If I did pem, err :=... then a new pem gets created inside the block and disappears when we exit the block scope.
I moved pem and err to be named return values and I think that cleans it up and works properly.